srv/salt/_modules/iscsi.py

```diff
@@ -425,20 +456,60 @@ def validate(lio_root):
     """
     targets_by_disk = {}
     for target in lio_root.targets:
+        tpg_auth_prev = None
+        tpg_acls_prev = None
         for tpg in target.tpgs:
+            tpg_auth = {tpg.get_attribute('generate_node_acls'),
+                        tpg.get_attribute('authentication'),
+                        tpg.chap_userid,
+                        tpg.chap_password,
+                        tpg.chap_mutual_userid,
+                        tpg.chap_mutual_password,
+                        tpg.authenticate_target}
+            if (tpg.get_attribute('generate_node_acls') == '0' and
+              tpg.chap_userid is not None and tpg.chap_password is not None):
+                raise Exception(
+                    'Unsupported LIO configuration: concurrent ACL and TPG '
+                    'based authentication for target ({}).'.format(target.wwn))
+            if tpg_auth_prev is None:
+                tpg_auth_prev = tpg_auth
+            elif tpg_auth_prev != tpg_auth:
+                raise Exception(
+                    'Unsupported LIO configuration: Authentication settings '
+                    'differ between TPGs for target ({}). Switch lrbd to target'
+                    'based authentication to proceed.'.format(target.wwn))
+            if tpg_acls_prev is None:
+                tpg_acls_prev = tpg.node_acls
+            elif tpg_acls_prev != tpg.node_acls:
```

Comment:
`tpg.node_acls` is a list generator and cannot be use directly with a comparison operator like `==`.
Change this code to:
```python
    tpg_acls_prev = list(tpg.node_acls)
elif tpg_acls_prev != list(tpg.node_acls):
```